lucene/core/src/java/org/apache/lucene/search/Boolean2ScorerSupplier.java

```diff
@@ -118,6 +118,21 @@ private Scorer getInternal(long leadCost) throws IOException {
           leadCost);
     }
 
+    // pure two terms disjunction
+    if (scoreMode == ScoreMode.TOP_SCORES
+        && minShouldMatch == 0
```

Comment:
`minShouldMatch = 1` also qualifies

```suggestion
        && minShouldMatch <= 1
```

Reply:
Updated.